astropy/uncertainty/tests/test_distribution.py

```diff
@@ -15,30 +17,42 @@
     SMAD_FACTOR = 1 / norm.ppf(0.75)
 
 
-def test_numpy_init():
-    # Test that we can initialize directly from a Numpy array
-    rates = np.array([1, 5, 30, 400])[:, np.newaxis]
-    parr = np.random.poisson(rates, (4, 1000))
-    Distribution(parr)
-
-
-def test_numpy_init_T():
-    rates = np.array([1, 5, 30, 400])
-    parr = np.random.poisson(rates, (1000, 4))
-    Distribution(parr.T)
-
-
-def test_quantity_init():
-    # Test that we can initialize directly from a Quantity
-    pq = np.random.poisson(np.array([1, 5, 30, 400])[:, np.newaxis],
-                           (4, 1000)) * u.ct
-    Distribution(pq)
-
-
-def test_quantity_init_T():
-    # Test that we can initialize directly from a Quantity
-    pq = np.random.poisson(np.array([1, 5, 30, 400]), (1000, 4)) * u.ct
-    Distribution(pq.T)
+class TestInit:
+    @classmethod
+    def setup_class(self):
+        self.rates = np.array([1, 5, 30, 400])[:, np.newaxis]
+        self.parr = np.random.poisson(self.rates, (4, 1000))
+        self.parr_t = np.random.poisson(self.rates.squeeze(), (1000, 4))
+
+    def test_numpy_init(self):
+        # Test that we can initialize directly from a Numpy array
+        Distribution(self.parr)
+
+    def test_numpy_init_T(self):
```

Comment:
Is the point here that we're instantiating from a view object? or is there something special about the transpose?

Reply:
Here, I am really just collecting the function tests that @eteq put in originally in a class...  I think the original reason is that a view as a new `.dtype` gets taken and one better be sure that that is possible.  It would be better to actually check that the output has the right shape and so on, but maybe I can be excused because this is existing code :smiley_cat:

Reply:
GitHub needs a :shipit: (shipit) reaction. The rocket just doesn't suffice.